languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -34110,6 +34110,125 @@ though, and has since been slightly modified)-->
     </rule>
   </category>
   <category id="CAT_TYPOGRAPHIE" name="Typographie" type="typographical">
+      <rulegroup id="VIRGULE_DEBUT_DE_PHRASE" name="virgule debut de phrase" default="temp_off">
+          <rule>
+              <pattern>
+                      <token postag="SENT_START"/>
+                  <marker>
+                      <token regexp="yes">cependant|néanmoins|pourtant|toutefois</token>
```

Comment:
You'll have seen in the rule editor that there are times when, for whatever reason, the user writes just the word "Toutefois", with or without a period afterwards.

For that reason, I'd put two exceptions in here: `<exception postag="SENT_END"/><exception scope="next" postag="SENT_END"/>`.

How about shortened sentences like "Toutefois pas ici."? Should we maybe exclude those, too?

Reply:
About sentences like "Toutefois pas ici." I'd say the comma is still needed. After expressions like "toutefois", I'd always put a comma in the sentence except for "Toutefois..." as the speaker is thinking and the sentence is ending.